cmd/erasure-metadata.go

```diff
@@ -357,24 +358,20 @@ func findFileInfoInQuorum(ctx context.Context, metaArr []FileInfo, modTime time.
 					found = true
 				}
 				succModTimeMap[metaArr[i].SuccessorModTime]++
+				numVersionsMap[metaArr[i].NumVersions]++
 			}
 		}
 	}
-	var succModTime time.Time
-	var smodTimeQuorum bool
-	for smodTime, count := range succModTimeMap {
-		if count >= quorum {
-			smodTimeQuorum = true
-			succModTime = smodTime
-			break
-		}
-	}
 
 	if found {
-		if smodTimeQuorum {
+		// Update candidate FileInfo with succModTime and numVersions in quorum when available
+		if succModTime, ok := succModTimeMap.GetValueWithQuorum(quorum); ok {
 			candidate.SuccessorModTime = succModTime
 			candidate.IsLatest = succModTime.IsZero()
 		}
+		if numVersions, ok := numVersionsMap.GetValueWithQuorum(quorum); ok {
+			candidate.NumVersions = numVersions
+		}
 		return candidate, nil
```

Comment:
shouldn't we agree on NumVersions, SuccessorModTime ? having them separate would lead to different outcomes? 

It is ever possible they agreed on NumVersions not agree on SUccessorModTime?

Reply:
Good point. In practice, there will be only a single quorum so it should not have different outcomes. It is possible that there are no good outcomes, i.e we may not have a quorum about say a second version, which affects both SuccessorModTime and NumVersions. Since it is not possible to have different values anyway, I shall modify the PR to look for quorum for the combination of SuccessorModTime and NumVersions when available.